parsa-rajabi-ubc/Home-Together-Canada
# HTC-774/messaging connect backend

# [#774 ](https://github.com/rachellegelden/Home-Together-Canada/issues/774)

## Summary
Connect backend to frontend.
Update: display delete account in both message container and full conversation between 2 users. Also, if a user is deleted, then people will cannot send any messages to him (sendMessage will not display).

## Relevant Motivation & Context
We need to connect backend to frontend. 
Note: 
- It is not a real time messaging system, you have to go to another nav then come back to see the new messages.
- After send a message, the text area is not empty, will have the value message content that you just send.

## Testing Instructions
1. Create 3 users (userA, userB and userC) in post postman 
2. In postman login as userA , and create some messages that send to userB and userC
3. logout as userA
4. repeat step 2 and 3 for userB and userC
5. login in as userA in http://localhost:3002/, then go to searchmember profile, search for B and C. And send a message to both of them.
6. As userA in http://localhost:3002/, go to messaging
7. Clicks on both conversation cards to go in to the conversation detail between 2 users
8. View and check the output
9. Create a message in textarea, and hit send button
10. Go to main page or other nav then come back to messaging, the message you just sent should display in both "converation card" and "conversation detail"
11. Logout as all users in postman then login as userB
12. delete userB account, refresh in http://localhost:3002/, as userA
13. Go to messaging and trying to send the message to userB again it will show the alert of that user account is deleted.
14. Note that sendmessage component is no longer rendering 
15. As userA search for userC, go to send a message, don't submit
16. In postman, delete userC
17. Try to send the message now (Should get a alert that userC has been deleted)
18. Create 2 new members userB and userC
19. Repeat step5 
20. Go to Messaging, you should see 4 conversations. 2 for deleted account, 2 for new users.
21. Repeat step8 

## Developer checklist prior to opening this pull request:

## Reviewer
